_posts/2019-06-25-single-gpu-cupy-benchmarks.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: post
+title: Single-GPU CuPy Benchmarks
+author: Peter Andreas Entschev
+tags: [CuPy, SVD]
+theme: twitter
+---
+{% include JB/setup %}
+
+
+Summary
+-------
+
+Array operations with GPUs can provide considerable speedups over CPU computing,
+but the amount of speedup varies greatly depending on the operation. The intent
+of this blog post is to benchmark CuPy performance for various different
+operations. We can definitely plug Dask in to enable multi-GPU performance gains,
+as discussed
+[in this post from March](https://blog.dask.org/2019/03/18/dask-nep18), but here
+we will only look at individual performance for single-GPU CuPy.
+
+
+Hardware and Software Setup
+---------------------------
+
+Before going any further, assume the following hardware and software is
+utilized for all performance results described in this post:
+
+* System: NVIDIA DGX-1
+* CPU: 2x Intel Xeon E5-2698 v4 @ 2.20GHz
+* Main memory: 1 TB
+* GPU: NVIDIA Tesla V100 32 GB
+* Python 3.7.3
+* NumPy 1.16.4
+* Intel MKL 2019.4.243
+* CuPy 6.1.0
+* CUDA Toolkit 9.2
+
+
+General Performance
+-------------------
+
+I have generated a greph comprising various operations. Most of them perform
+well on a GPU using CuPy out of the box. See the graph below:
+
+<img src="/images/single-gpu-cupy-benchmarks.png">
```

Comment:
If you wanted to embed the Vegalite JSON output itself you might want to look at this section here: https://github.com/dask/dask-blog/pull/32/files#diff-eb9896a7fb2ebd01e20ad59828c95e83R45

Reply:
Also, this image here isn't added.  I'd be happy to copy-paste my version over though if that works for you (it plays nicely with the formatting of our layout).

Reply:
Thanks for the link, I've updated it now.